api/src/main/java/org/openmrs/module/htmlformentryui/element/ObsFromFragmentElement.java

```diff
@@ -163,12 +165,16 @@ protected Object convertToType(String val) {
 		}
 		if (dataType.isDateTime() || dataType.isTime()) {
 			try {
-				DateFormat df = new SimpleDateFormat(WebConstants.DATE_FORMAT_DATETIME);
-				df.setLenient(false);
-				if (dataType.isDateTime() && uiUtils.convertTimezones() && uiUtils.getClientTimezone() != null) {
-					df.setTimeZone(TimeZone.getTimeZone(uiUtils.getClientTimezone()));
+				// Datetimepicker can send one of 2 formats "yyyy-MM-dd HH:mm:ss" / "yyyy-MM-dd'T'HH:mm:ss.SSSZ"
```

Comment:
What are the conditions which lead to the widget sending either of the two formats?